subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -1064,71 +1053,18 @@ static void adv_iso_chm_complete_commit(struct lll_adv_iso *lll_iso)
 	lll_adv_sync_data_enqueue(lll_sync, ter_idx);
 }
 
-static void mfy_iso_offset_get(void *param)
+void ull_adv_iso_lll_biginfo_fill(struct pdu_adv *pdu, struct lll_adv_sync *lll_sync)
 {
-	struct lll_adv_sync *lll_sync;
-	struct ll_adv_sync_set *sync;
 	struct lll_adv_iso *lll_iso;
-	uint32_t ticks_to_expire;
 	struct pdu_big_info *bi;
-	uint32_t ticks_current;
 	uint64_t payload_count;
-	struct pdu_adv *pdu;
-	uint8_t ticker_id;
-	uint16_t lazy;
-	uint8_t retry;
-	uint8_t id;
-
-	sync = param;
-	lll_sync = &sync->lll;
-	lll_iso = lll_sync->iso;
-	ticker_id = TICKER_ID_ADV_ISO_BASE + lll_iso->handle;
 
-	id = TICKER_NULL;
-	ticks_to_expire = 0U;
-	ticks_current = 0U;
-	retry = 4U;
-	do {
-		uint32_t volatile ret_cb;
-		uint32_t ticks_previous;
-		uint32_t ret;
-		bool success;
-
-		ticks_previous = ticks_current;
-
-		ret_cb = TICKER_STATUS_BUSY;
-		ret = ticker_next_slot_get_ext(TICKER_INSTANCE_ID_CTLR,
-					       TICKER_USER_ID_ULL_LOW,
-					       &id, &ticks_current,
-					       &ticks_to_expire, NULL, &lazy,
-					       NULL, NULL,
-					       ticker_op_cb, (void *)&ret_cb);
-		if (ret == TICKER_STATUS_BUSY) {
-			/* Busy wait until Ticker Job is enabled after any Radio
-			 * event is done using the Radio hardware. Ticker Job
-			 * ISR is disabled during Radio events in LOW_LAT
-			 * feature to avoid Radio ISR latencies.
-			 */
-			while (ret_cb == TICKER_STATUS_BUSY) {
-				ticker_job_sched(TICKER_INSTANCE_ID_CTLR,
-						 TICKER_USER_ID_ULL_LOW);
-			}
-		}
-
-		success = (ret_cb == TICKER_STATUS_SUCCESS);
-		LL_ASSERT(success);
-
-		LL_ASSERT((ticks_current == ticks_previous) || retry--);
-
-		LL_ASSERT(id != TICKER_NULL);
-	} while (id != ticker_id);
+	lll_iso = lll_sync->iso;
 
-	payload_count = lll_iso->payload_count + ((lll_iso->latency_prepare +
-						   lazy) * lll_iso->bn);
+	payload_count = lll_iso->payload_count + (lll_iso->iso_lazy * lll_iso->bn);
```

Comment:
Yes, it is set in the ticker_cb based on the other_expire_info that the ticker provides (in ull_adv_sync.c)

Reply:
`lll_iso->latency_prepare`  and `lll_iso->iso_lazy` should both be used, see implementation of `ull_conn_event_counter()` how it handles prepare callback that are in prepare pipeline, i.e. latency accumulated with reference get incremented.

Reply:
> `lll_iso->latency_prepare` and `lll_iso->iso_lazy` should both be used, see implementation of `ull_conn_event_counter()` how it handles prepare callback that are in prepare pipeline, i.e. latency accumulated with reference get incremented.

I don't know how latency_prepare in `lll_conn` is used; But latency_prepare in `lll_adv_iso` is set to `lazy + 1` in `prepare` and then set back to 0 `prepare_cb_common` (see lll_adv_iso.c) - ie. it is just a snapshot of the lazy parameter. So if you include latency_prepare in the calculation of payload_count, you will effectively be calculating it as `lll_iso->payload_count + ((lll_iso->iso_lazy*2 + 1) * lll_iso->bn);` (when called in between `prepare` and `prepare_common`), which looks very wrong to me.

Reply:
`latency_prepare` in all state/role LLL contexts accumulate `lazy` after `ticker_cb` has executed and the prepare callbacks wait (preempt timeout duration) in the pipeline until `prepare_cb` is executed. During this wait, prepare can be aborted (removed from pipeline) and hence `latency_prepare` accumulates all `lazy` value until subsequent `prepare_cb` gets executed.

That said, I will test this PR and get back this week.

Reply:
> `latency_prepare` in all state/role LLL contexts accumulate `lazy` after `ticker_cb` has executed and the prepare callbacks wait (preempt timeout duration) in the pipeline until `prepare_cb` is executed. During this wait, prepare can be aborted (removed from pipeline) and hence `latency_prepare` accumulates all `lazy` value until subsequent `prepare_cb` gets executed.

Ah - yes I missed the `+=` in `prepare`. But since it is incremented by `lazy + 1` it is problematic to include this in the payload_count calculation... I'll have a think about how this can be handled.

Reply:
> since it is incremented by lazy + 1 it is problematic to include this in the payload_count calculation

Yes, `prepare_cb` execution depends on requirement to preempt active/current radio event or immediately execute. Hence, I assume the special handling in `ull_conn_event_counter()` used by LLCP implementation.

Reply:
Implemented something similar to ´ull_conn_event_counter()´ to account for being in a post-prepare situation; Looks like it is working.

Reply:
Tested Broadcast ISO samples (https://github.com/zephyrproject-rtos/zephyr/pull/55856)  and Central Multiple CIS support BabbleSIM (https://github.com/zephyrproject-rtos/zephyr/pull/54657/commits/32a128d55421ebe89fe18ffba158acd3d53edb28). The changes in this PR show no regression in these minimal testing or in the CI tests.